src/Appender/RollingFileAppender.cs

```diff
@@ -1147,8 +1148,20 @@ override public void ActivateOptions()
 			}
 
 #if !NETCF
-			// initialize the mutex that is used to lock rolling
-			m_mutexForRolling = new Mutex(false, m_baseFileName.Replace("\\", "_").Replace(":", "_").Replace("/", "_"));
+			if (m_mutexForRolling == null)
+			{
+				// initialize the mutex that is used to lock rolling
+				string mutexFriendlyFilename = m_baseFileName.Replace("\\", "_").Replace(":", "_").Replace("/", "_");
```

Comment:
Could this dark magic be moved into the `SafeCreateMutex()`, such that it exists only once?

Reply:
in the new method, is it okay, to replace all `Path.GetInvalidFileNameChars()` not just `\/:`?

Reply:
I am unsure whether the application still behaves well. Mutex names are not a path or a file name.

This exists to guess a mostly unique mutex name from the filename. This way another rolling file appender configured to roll another filename can happily roll at the same time.

So long this is preserved the code can do anything. Even md5 hashing would be ok.